widgets/grid/org.eclipse.nebula.widgets.grid/src/org/eclipse/nebula/widgets/grid/Grid.java

```diff
@@ -5447,24 +5450,29 @@ private void onPaint(final PaintEvent e) {
 					x += rowHeaderWidth;
 				}
 
-				emptyCellRenderer.setBounds(x, y, getClientArea().width - x, getItemHeight());
+				emptyCellRenderer.setBounds(x, y, clientArea.width - x, getItemHeight());
 				emptyCellRenderer.setFocus(false);
 				emptyCellRenderer.setSelected(false);
 				emptyCellRenderer.setRow(i + 1);
 
 				for (final GridColumn column : displayOrderedColumns) {
 
 					if (column.isVisible()) {
-						emptyCellRenderer.setBounds(x, y, column.getWidth(), getItemHeight());
-						emptyCellRenderer.setColumn(indexOf(column));
-						emptyCellRenderer.paint(e.gc, this);
-
-						x += column.getWidth();
+						int width = column.width;
+						if(x + width >= 0) {
+							emptyCellRenderer.setBounds(x, y, width, getItemHeight());

```

Comment:
you could move the getItemHeight() out of the loop or access it directly to by pass the check